### PR TITLE
fix timeline

### PR DESCRIPTION
**Pull request by koilebeit**
# Pull request

## Proposed changes

solves #165 
## Types of changes

- [ ] New feature (non-breaking change which adds functionality).
- [ ] Enhancement (non-breaking change which enhances functionality)
- [x] Bug Fix (non-breaking change which fixes an issue).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist

- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Summary by CodeRabbit

- **New Features**
	- Improved timeline layout with sorted year navigation and structured year display.
	- Enhanced sorting mechanism for date-related strings, providing detailed and structured output.
	- Introduced a method to filter items by year using regex for better data handling.

- **Bug Fixes**
	- Corrected filtering of timeline entries to ensure accurate display based on year selection.

### CODE REVIEW

File: _plugins/sort_edtf.rb [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
+                         else
+                           str.sub(/^0+/, '')  # Remove leading zeros for positive values
+                         end
+
+        # Return a hash with numeric, original, and display_format
+        { numeric: numeric_value, original: str, display_format: display_format }
+      end
+
+      # Sort by the numeric representation
+      sorted_dates = parsed_dates.sort_by { |date| date[:numeric] }
```
[the remark this answers is not in the document]
Comment (koilebeit) [reply _plugins/sort_edtf.rb:35]: in order to be able to sort and render the data, it must be converted to exact numerical values. this is therefore intentional

File: _plugins/sort_edtf.rb [hunk 2]
```diff
@@ -1,10 +1,36 @@
...
+
+      # Return an array of "display_format:original" strings
+      sorted_dates.map { |date| "#{date[:display_format]}:#{date[:original]}" }
```
[the remark this answers is not in the document]
Comment (koilebeit) [reply _plugins/sort_edtf.rb:38]: unfortunately liquid is limited in dealing with complex structures like objects, therefore concatenated strings are returned

File: _plugins/sort_edtf.rb [hunk 3]
```diff
@@ -1,10 +1,36 @@
...
-    module SortEDTF
-        def sort_edtf(array_of_strings)
-            sorted = array_of_strings.map { |str| str.gsub('X', '0') }
-            sorted.sort_by { |str| str[/\d+/].to_i }
-        end
+  module SortEDTF
+    def sort_edtf(array_of_strings)
+      # Parse each string into a hash with numeric, original, and display representations
+      parsed_dates = array_of_strings.map do |str|
+        cleaned_str = str.gsub(/[Xx]/, '0') # Replace X/x with 0 for numeric comparison
```
[the remark this answers is not in the document]
Comment (koilebeit) [reply _plugins/sort_edtf.rb]: that can be discussed. @maehr your opinion?

**Review by maehr — CHANGES_REQUESTED** [review]
Thanks for the PR. Unfortunately it does not build.
```bash
  Liquid Exception: Liquid error (line 13): Invalid date format: in /workspaces/forschung.stadtgeschichtebasel.ch/_layouts/timeline_edtf.html
                    ------------------------------------------------
      Jekyll 4.3.4   Please append `--trace` to the `build` command 
                     for any additional information or backtrace. 
                    ------------------------------------------------
/workspaces/forschung.stadtgeschichtebasel.ch/_plugins/sort_edtf.rb:19:in `block in sort_edtf': Liquid error (line 13): Invalid date format:  (Liquid::ArgumentError)
```

File: _layouts/timeline_edtf.html [hunk 4]
```diff
@@ -37,30 +38,33 @@
...
+      {% assign year = date | split: ':' %}
+      <tr id="y{{ year[1] }}">
         <th>
-          <h3>{{ year }}</h3>
+          <h3>{{ year[0] }}</h3>
         </th>
         <td>
           <div class="row">
-            {%- assign inYear = items | where_exp: 'item', 'item[field] contains year' -%}
+            {%- assign inYear = items | where_exp: 'item', 'item[field] contains year[1]' -%}
```
[the remark this answers is not in the document]
Comment (koilebeit) [reply _layouts/timeline_edtf.html]: https://github.com/Stadt-Geschichte-Basel/forschung.stadtgeschichtebasel.ch/pull/171/commits/d4996c005d1efd3fb153f873733e46ff86c85546

**Review by maehr — APPROVED** [review]
Thanks again. It seems to work perfectly!

File: _plugins/sort_edtf.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (maehr) [reply _plugins/sort_edtf.rb]: I like your solution better.

**Pull request merged**